### PR TITLE
Assign team to championship

### PR DESCRIPTION
**Pull request by Mickss**

### CODE REVIEW

**Review by tknowak — APPROVED** [review]

File: src/main/java/org/micks/champmaker/championships/ChampionshipController.java [hunk 1]
```diff
@@ -21,11 +24,19 @@ public class ChampionshipController {
...
+    @Autowired
+    private RegisterService registerService;
+
     @GetMapping("/{champId}")
     public ChampionshipDTO getChampionship(@PathVariable long champId) throws EntityNotFoundException {
         return championshipService.getChampionship(champId);
     }
 
+    @PutMapping(value = "/{champId}", consumes = MediaType.APPLICATION_JSON_VALUE)
+    public void registerTeamToChampionship(@PathVariable long champId, @RequestBody RegisterDTO registerDTO) {
```
Comment (kbnowak) [inline src/main/java/org/micks/champmaker/championships/ChampionshipController.java:36]: Is there any reason why we don't keep champId inside RegisterDTO?

File: src/main/java/org/micks/champmaker/register/RegisterDTO.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    }
+
+    public long getTeamId() {
+        return teamId;
+    }
+
+    public String getRegistrationDate() {
+        return registrationDate;
+    }
+}
```
Comment (kbnowak) [inline src/main/java/org/micks/champmaker/register/RegisterDTO.java:22]: add newline at the end

File: src/main/java/org/micks/champmaker/register/RegisterDTO.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package org.micks.champmaker.register;
+
+public class RegisterDTO {
```
Comment (kbnowak) [inline src/main/java/org/micks/champmaker/register/RegisterDTO.java:3]:
```suggestion
public class RegisterDto {
```
Leaving abbreviations in all capitals can lead to really bad names like ORMXMLSAMLConverter - but that would apply to all DTOs in the project 😄

File: src/main/java/org/micks/champmaker/register/RegisterService.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class RegisterService {
+
+    @Autowired
+    private RegisterRepository registerRepository;
+
+
```
Comment (kbnowak) [inline src/main/java/org/micks/champmaker/register/RegisterService.java:12]: remove double newline

File: src/main/java/org/micks/champmaker/championships/ChampionshipController.java (hunk 1 above)
Comment (tknowak) [reply src/main/java/org/micks/champmaker/championships/ChampionshipController.java:36]: Yes, champId is primary id, inside championshipController. The rest is data for assigning something (team in this case) to championship. I think in classic Rest every PUT request should have path id for resource we are changing.

**Pull request merged**